### ISSUE TITLE
brace-style ignores braces around class body

### ISSUE THREAD

**Issue by fasttime** [eslint/eslint#7608]
Not very sure about this. I'll post it here anyway so someone can have a look. Feel free to close if I'm horribly wrong.

Using this configuration:

```
{ 'brace-style': ['error', 'allman'] }
```

**What did you do? Please include the actual source code causing the issue.**

```js

class Foo {
}

```

**What did you expect to happen?**

According to Allman style rules, the opening brace should appear on a new line, and so the code above should trigger an error.

**What actually happened? Please include the actual, raw output from ESLint.**

No error.

**Comment by platinumazure** [comment]
@fasttime What ESLint version are you using? Can you please confirm you get the same results when using latest ESLint?

That said, looking at the rule source, I don't see any logic around class declarations, so it's very likely we just never implemented this for class declarations.

**Comment by platinumazure** [comment]
# Rule enhancement proposal

**What rule do you want to change?**
brace-style

**Does this change cause the rule to produce more or fewer warnings?**
More

**How will the change be implemented? (New option, new default behavior, etc.)?**
New default behavior, ideally (this is a missed use case)

**Please provide some example code that this change will affect:**

```js
/* eslint-env es6 */
/* eslint brace-style: ["error", "allman"] */

class Foo {
}
```

**What does the rule currently do for this code?**
No error

**What will the rule do after it's changed?**
Warn for incorrect brace style

Note: This applies to all brace styles, not just Allman.

I'll champion.

**Comment by not-an-aardvark** [comment]
I'm :+1: either way, but this seems like a bug to me, not an enhancement (especially given that we would fix it in a minor release and cause more errors to be reported).

**Comment by platinumazure** [comment]
I'll split the difference by using both labels, which we've used in the past for semver-minor bug fixes.

Need two more :+1: from team members before we can accept this issue.

**Comment by evilebottnawi** [comment]
@platinumazure what status? I can help (PR) :smile:

**Comment by platinumazure** [comment]
@evilebottnawi Really sorry for taking so long to get back to you.

I think @not-an-aardvark is right about this being a bug, so we can accept the issue right away and you may submit a PR whenever you have time! Just keep us posted if you're no longer able to, so we can get someone else to do so. Thanks!

**Comment by evilebottnawi** [comment]
@platinumazure have time on ending this week

**Comment by platinumazure** [comment]
@fasttime @evilebottnawi Either of you want to take a crack at this?

**Comment by not-an-aardvark** [comment]
I'm willing to work on this is no one else is doing it -- I haven't done it yet because I thought @evilebottnawi was working on it.

**Comment by fasttime** [comment]
@platinumazure I have zero experience working on the ESLint codebase, just some familiarity with parser logic and ASTs. I'll leave this one to @not-an-aardvark, but I'd be glad to help out another time.
